main/views.py

```diff
@@ -1,23 +1,77 @@
 from django.shortcuts import render, redirect
 from django.http import HttpResponse
 from .forms import GeolocatorStatsForm
+from .forms import GeolocatorStatsFormHistory
 from .models import GeolocatorStats
 import requests
+import re
 
+# HOW TO DEAL WITH ERROR MESSAGES 
+history = []
+#navbar, history, about, home
 def index(request):
```

Comment:
Consider having multiple functions for handling different cases.
if condition1:
  function1()
 elif condition2:
  function2()
else:
  functionElse()